### ISSUE TITLE
Nightly build failure building clang8-cuda10 in `KokkosSparse_spmv_bsrmatrix_impl.hpp"

### ISSUE THREAD

**Issue by cwpearson** [kokkos/kokkos-kernels#1306]
The functor is compiled with CUDA-specific code, and uses something like

```
__half *h;
*h = 0;
```

But, `operator=(int)` for `__half` is a `__device__` function in CUDA 10.

```
05:19:13  /home/projects/x86-64/cuda/10.0/include/cuda_fp16.hpp:197:24: note: 'operator=' declared here
05:19:13      __device__ __half &operator=(int val) { __x = __int2half_rn(val).__x; return *this; }
```

([jenkins](https://jenkins-son.sandia.gov/view/KokkosKernels/job/KokkosKernels_KokkosDev2_CLANG8_CUDA10_cpp14/lastFailedBuild/console))

**Comment by ndellingwood** [comment]
Jenkins errors reported duplicate of #1209

**Comment by ndellingwood** [comment]
@cwpearson thanks for the PR #1307, that resolved some clang+cuda compilation errors with the nightly builds. I tested the build on kokkos-dev-2 and I'm now seeing some test failures in the `sparse_spmv_tensor_core` tests, I'll post a new issue with details

**Comment by cwpearson** [comment]
Fixed in #1307
